### PR TITLE
Creating test Kubernetes clusters with Kind

### PR DESCRIPTION
**Pull request by mcasperson**
# About

This is a post about configuring Kind with Octopus for local development in Kubernetes.

# Pre-requisites

- [ * ] The draft is complete and this post is ready to be reviewed.
- [ * ] I'm familiar with the [writing for Octopus TL;DR](https://style.octopus.com/writing-at-octopus-tldr).
- [ * ] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://style.octopus.com/writing-at-octopus-tldr#common-validation-errors) and none of those apply.
- [ * ] The images I've included follow the [rules for working with images](https://style.octopus.com/images#rules-for-screenshots).

## Publication date

This can be published at any time.

### CODE REVIEW

**Review by robpearson — APPROVED** [review]
This is a wonderful post Matt. I really enjoyed reading it. 😁 

My only comment is if this is applicable to windows and non-windows folks? If it will run on Linux or macOS, can we state that upfront? If not, do the scripts need bit changes to support non-windows.

Best case scenario is we show how to do this on both Windows and Mac (or linux) so we broaden our appeal to both .NET and Java folks etc. Any thoughts?

**Comment by mcasperson** [comment]
> My only comment is if this is applicable to windows and non-windows folks? If it will run on Linux or macOS, can we state that upfront? If not, do the scripts need bit changes to support non-windows.

By running the deployment in WSL2 the example is completely Linux based. The sample scripts are bash scripts, and all the examples use the Linux tentacles. I have added sample scripts in powreshell for Windows users in the main blog, but the main thrust of this blog targets Linux users.

> Best case scenario is we show how to do this on both Windows and Mac (or linux) so we broaden our appeal to both .NET and Java folks etc. Any thoughts?

The main aim here was to speak to a Linux audience, which is why the examples were all done in WSL2. I don't think there is much value in redoing this recording to target windows or macos as Kubernetes on Windows is very niche and macos would be so similar to Linux as to be almost direcly applicable. I also don't have a mac that I can record videos on, which would make doing a MacOS targeted example difficult.

File: blog/2020-09/testing-with-kind/index.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+---
+title: Creating test Kubernetes clusters with Kind
+description: Learn how to integrate a test cluster created with Kind with Octopus
+author: matthew.casperson@octopus.com
+visibility: private
+published: 2999-01-01
```
Comment (wordlee) [inline blog/2020-09/testing-with-kind/index.md:6]:
```suggestion
published: 2020-09-28
```

File: blog/2020-09/testing-with-kind/index.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
+---
+title: Creating test Kubernetes clusters with Kind
+description: Learn how to integrate a test cluster created with Kind with Octopus
+author: matthew.casperson@octopus.com
+visibility: private
```
Comment (wordlee) [inline blog/2020-09/testing-with-kind/index.md:5]:
```suggestion
visibility: public
```

File: blog/2020-09/testing-with-kind/index.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
+---
+title: Creating test Kubernetes clusters with Kind
+description: Learn how to integrate a test cluster created with Kind with Octopus
+author: matthew.casperson@octopus.com
+visibility: public
+published: 2020-09-28
+metaImage:
```
Comment (wordlee) [inline blog/2020-09/testing-with-kind/index.md:7]:
```suggestion
metaImage: kubernetes-in-docker.png
```

File: blog/2020-09/testing-with-kind/index.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
+---
+title: Creating test Kubernetes clusters with Kind
+description: Learn how to integrate a test cluster created with Kind with Octopus
+author: matthew.casperson@octopus.com
+visibility: public
+published: 2020-09-28
+metaImage: kubernetes-in-docker.png
+bannerImage:
```
Comment (wordlee) [inline blog/2020-09/testing-with-kind/index.md:8]:
```suggestion
bannerImage: kubernetes-in-docker.png
```

File: blog/2020-09/testing-with-kind/index.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
+---
+title: Creating test Kubernetes clusters with Kind
+description: Learn how to integrate a test cluster created with Kind with Octopus
+author: matthew.casperson@octopus.com
+visibility: public
+published: 2020-09-28
+metaImage: kubernetes-in-docker.png
+bannerImage: kubernetes-in-docker.png
+tags:
+ - Octopus
```
Comment (wordlee) [inline blog/2020-09/testing-with-kind/index.md:10]:
```suggestion
 - DevOps
 - Kubernetes
```

**Pull request merged**